### ISSUE TITLE
v3.6.0 does not find all Outlook mailboxes

### ISSUE THREAD

**Issue by GruberMarkus** [GruberMarkus/Set-OutlookSignatures#75]
### Discussed in https://github.com/GruberMarkus/Set-OutlookSignatures/discussions/74

<div type='discussions-op-text'>

<sup>Originally posted by **KoenBouten** April 28, 2023</sup>
Hello Markus,

Here you have the text file with the output of the 3.6.0 script which you asked for last time.

But there was one thing I found in the meantime:

3.3.0 = line 748
3.6.0 = line 744
The script displays things in a different order/way at this part.
(domainD.local is a one-way-domain-trust & domainH.nl is a forrest-trust)

I hope this will be enough information.

[Script output V3.6.txt](https://github.com/GruberMarkus/Set-OutlookSignatures/files/11352226/Script.output.V3.6.txt)
[Script OutputV3.3.txt](https://github.com/GruberMarkus/Set-OutlookSignatures/files/11352255/Script.OutputV3.3.txt)
</div>

**Comment by GruberMarkus** [comment]
So we can just ignore issues A and B.

Let's focus on issue C now.

Your maibox is running in a resource forest scenario: You have a user in AD forest A, Exchange is running in AD forest D. The user from forest A is configured as linked account (a.k.a. external master account) on a mailbox in AD forest D.

This configuration is nothing uncommon, and it is supported by Set-OutlookSignatures (other products do not have this feature).

Microsoft does not required to set the mail attribute on the user in AD forest A, and you have not done this. Set-OutlookSignatures logs this with this text:
```
Get AD properties of currently logged in user and assigned manager @2023-04-28T09:32:44+02:00@
  Currently logged in user
    User's mailbox not found in Outlook profiles, but Outlook Web signature and/or OOF message should be set. Add dummy mailbox entry.
      User does not have mail attribute configured
    CN=Koen Bouten,OU=Users,OU=domainA,DC=domainA,DC=local
```

The mail attribute of the currently logged in user is important for Set-OutlookSignatures, as the mailbox of the currently logged in user has highter priority than other mailboxes. As the mail attribute is not set, Set-OutlookSignatures tries to find the mail address by searching for a mailbox in AD, where the user from AD forest A is set as linked account. The search is successful:
```
Sort mailbox list: User's primary mailbox, mailboxes in default Outlook profile, others @2023-04-28T09:32:44+02:00@
  AD mail attribute of currently logged in user is empty, searching msExchMasterAccountSid
    One matching primary mailbox found: k.bouten@domainA.com
  Mailbox priority (highest to lowest)
```

The lines following are a symptom of the bug in v3.6.0 which is already fixed in the yet unreleased version v4.0.0.:
```
VERBOSE:   Outlook mailbox display sort order is not yet defined or does not yet contain all found mail addresses. Falling back to sorting by time mailboxes
 have been added.
    k.bouten@domainA.com
VERBOSE:       Outlook profile 'Outlook'
VERBOSE:         HKCU\Software\Microsoft\Office\16.0\Outlook\Profiles\Outlook\9375CFF0413111d3B88A00104B2A6676\00000002
VERBOSE:         /o=First Organization/ou=Exchange Administrative Group (FYDIBOHF23SPDLT)/cn=Recipients/cn=cdb645573cba43a8a87b3ccf02b499bc-Koen
```


All in all, I would say issue C is not a problem, but a configuration that is
- supported by Microsoft
- supportet by Set-OutlookSignatures
- has no impact on how Set-OutlookSignatures works
and is therefore not a bug.

I think, we can close this issue. Do you agree?

**Comment by GruberMarkus** [comment]
My pleasure!
